### DISCUSSION TITLE
Solution with C# project and VB Project

### DISCUSSION THREAD

**Discussion by birojnayak** [dotnet/roslyn#67788] [Q&A] [answered]
I am using CSharpCompilation.Create and VisualBasicCompilation.Create to generate compilation unit . This works well when I have all projects under a solution with all dependent projects in either C# or VB.  The problem I am running into is if I have a mix projects meaning I have C# project which is dependent on VB project. I can generate the compilation unit from VB project but I can't pass that as a metadata reference which generating the C# compilation unit. As this [here](https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L388) will throw error.

Any alternative or better ideas ?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Correct. This is not a supported scenario. You will have to emit the vb project to metadata (see Compilation.EmitXxx) and then reference that from the c# project.
